library/ReplicatedTableEntity.cs

```diff
@@ -100,24 +101,112 @@ public ReplicatedTableEntity(string partitionKey, string rowKey)
             this.RowKey = rowKey;
         }
 
+        public ReplicatedTableEntity(ReplicatedTableEntity entity)
+            : this(entity.PartitionKey, entity.RowKey)
+        {
+            this.CopyFrom(entity);
+        }
+
+        public ReplicatedTableEntity(TableEntity entity)
+            : this(entity.PartitionKey, entity.RowKey)
+        {
+            if (entity == null)
+            {
+                throw new ArgumentNullException(nameof(entity));
+            }
+
+            this._rtable_RowLock = bool.Parse(entity["_rtable_RowLock"].ToString());
+            this._rtable_Version = long.Parse(entity["_rtable_Version"].ToString());
+            this._rtable_Tombstone = bool.Parse(entity["_rtable_Tombstone"].ToString());
+            this._rtable_ViewId = long.Parse(entity["_rtable_ViewId"].ToString());
+            this._rtable_Operation = entity["_rtable_Operation"].ToString();
+            this._rtable_BatchId = Guid.Parse(entity["_rtable_BatchId"].ToString());
+            this._rtable_LockAcquisition = DateTimeOffset.Parse(entity["_rtable_LockAcquisition"].ToString());
+
+            this.Properties = new Dictionary<string, object>();
+            foreach (var key in entity.Keys)
+            {
+                if (key.Equals("Properties", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    continue;
+                }
+                this.Properties.Add(key, entity[key]);
+            }
+        }
+
         /// <summary>
-        /// Deserializes this <see cref="DynamicTableEntity"/> instance using the specified <see cref="Dictionary{TKey,TValue}"/> of property names to values of type <see cref="EntityProperty"/>.
+        /// Deserializes this <see cref="DynamicTableEntity"/> instance using the specified <see cref="Dictionary{TKey,TValue}"/> of property names to values of type <see cref="object"/>.
         /// </summary>
-        /// <param name="properties">A collection containing the <see cref="Dictionary{TKey,TValue}"/> of string property names mapped to values of type <see cref="EntityProperty"/> to store in this <see cref="DynamicTableEntity"/> instance.</param>
-        /// <param name="operationContext">An <see cref="OperationContext"/> object used to track the execution of the operation.</param>
-        public virtual void  ReadEntity(IDictionary<string, EntityProperty> properties, OperationContext operationContext)
+        /// <param name="properties">A collection containing the <see cref="Dictionary{TKey,TValue}"/> of string property names mapped to values of type <see cref="object"/> to store in this <see cref="DynamicTableEntity"/> instance.</param>
+        public virtual void ReadEntity(IDictionary<string, object> properties)
```

Comment:
Need more context on this change, looks like we copied from some code from the previous library

Reply:
Copied code from TableEntity.ReflectionRead and TableEntity.ReflectionWrite
[TableEntityDecompiled.txt](https://github.com/Azure/rtable/files/14488429/TableEntityDecompiled.txt)